src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -0,0 +1,176 @@
+        shoulderTalonFX.config_kF(0, POSITION_KF, PID_CONFIG_TIMEOUT_MS);
+        shoulderTalonFX.config_kP(0, POSITION_KP, PID_CONFIG_TIMEOUT_MS);
+        shoulderTalonFX.config_kI(0, POSITION_KI, PID_CONFIG_TIMEOUT_MS);
+        shoulderTalonFX.config_IntegralZone(0, 1000);
+        shoulderTalonFX.config_kD(0, POSITION_KD, PID_CONFIG_TIMEOUT_MS);
+
+        reachTalonFX.config_kF(0, POSITION_KF, PID_CONFIG_TIMEOUT_MS);
+        reachTalonFX.config_kP(0, POSITION_KP, PID_CONFIG_TIMEOUT_MS);
+        reachTalonFX.config_kI(0, POSITION_KI, PID_CONFIG_TIMEOUT_MS);
+        reachTalonFX.config_IntegralZone(0, 1000);
+        reachTalonFX.config_kD(0, POSITION_KD, PID_CONFIG_TIMEOUT_MS);
+
+        reachTalonFX.setNeutralMode(NeutralMode.Brake);
+        shoulderTalonFX.setNeutralMode(NeutralMode.Brake);
+
+        reachTalonFX.configNeutralDeadband(MOTOR_NEUTRAL_DEADBAND);
+        shoulderTalonFX.configNeutralDeadband(MOTOR_NEUTRAL_DEADBAND);
+    }
+
+    @Override
+    public void periodic() {
+        moveToPosition(x,y);
+    }
+
+    /**
+     * Rotate the arm to a specific angle.
+     * @param angle - the angle in degrees
+     */
+    public void rotateTo(double angle) {
+        angle = Math.max(Math.min(angle, MAX_ANGLE), MIN_ANGLE);
+        double encoderUnits = angle * SHOULDER_ENCODER_UNITS_PER_DEGREE;
+        shoulderTalonFX.set(TalonFXControlMode.Position, encoderUnits);
+    }
+
+    /**
+     * Extends arm to a given amount of inches.
+     * @param length - the length to extend/retract to in inches
+     */
+    public void extendTo(double length) {
+        double encoderUnits = length * REACH_ENCODER_UNITS_PER_INCH;
+        reachTalonFX.set(TalonFXControlMode.Position, encoderUnits);
+    }
+
+    /**
+     * Moves the arm to a given position in space relative to the base.
+     * 
+     * @param targetX - position in inches
+     * @param targetY - position in inches
+     */
+    public void moveToPosition(double targetX, double targetY) {
+        double xDist = targetX - ARM_SHOULDER_X_INCHES;
+        double yDist = targetY - ARM_SHOULDER_Y_INCHES;
+        double length = Math.sqrt(Math.pow(xDist, 2) + Math.pow(yDist, 2));
+        double theta = Math.atan2(yDist, xDist);
+
+        rotateTo(theta);
+        extendTo(length);
+    }
+
+    /**
+     * Sets the internal position of the arm.
+     * @param xPos - position in inches
+     * @param yPos - position in inches
+     */
+    public void setPosition(double xPos, double yPos) {
+        x = xPos;
+        y = yPos;
+    }
+
+    public void adjustPosition(double xPercentage, double yPercentage) {
+        x += ARM_X_DELTA_MODIFIER * xPercentage;
+        y += ARM_Y_DELTA_MODIFIER * yPercentage;
+    }
+
+    public double encoderToDegrees(double encoderUnits) {
+        return encoderUnits / ENCODER_UNITS_PER_REVOLUTION * DEGREES_PER_REVOLUTION;
+    }
+
+    public void zeroEncoders() {
+        shoulderTalonFX.setSelectedSensorPosition(0);
```

Comment:
Let's split this into shoulder zero and reach zero

Reply:
It has been split